var/spack/repos/builtin/packages/mxnet/package.py

```diff
@@ -0,0 +1,101 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Mxnet(MakefilePackage):
+    """MXNet is a deep learning framework
+    designed for both efficiency and flexibility."""
+
+    homepage = "http://mxnet.io"
+    url      = "https://github.com/apache/incubator-mxnet/archive/0.10.0.post2.tar.gz"
+
+    version('0.10.0.post2',  '7819d511cf4a6efad681e6662fa966e4')
+    version('0.10.0.post1',  '16d540f407cd22285555b3ab22040032',
+       	    url="https://github.com/apache/incubator-mxnet/archive/v0.10.0.post1.tar.gz")
+    version('0.10.0', '2d0c83c33eda729932d620cca3078826',
+       	    url="https://github.com/apache/incubator-mxnet/archive/v0.10.0.tar.gz")
```

Comment:
You don't need explicit URLs for these versions, Spack is perfectly capable of extrapolating the correct URL:
```
$ spack url parse https://github.com/apache/incubator-mxnet/archive/0.10.0.post2.tar.gz
==> Parsing URL: https://github.com/apache/incubator-mxnet/archive/0.10.0.post2.tar.gz

==> Matched version regex  5: r'^[a-zA-Z+.]*(\\d[\\da-zA-Z.]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/apache/incubator-mxnet/archive/0.10.0.post2.tar.gz
                              ---------------         ~~~~~~~~~~~~
    name:    incubator-mxnet
    version: 0.10.0.post2

==> Substituting version 9.9.9b:
    https://github.com/apache/incubator-mxnet/archive/9.9.9b.tar.gz
                              ---------------         ~~~~~~
```